controllers/machine_controller_node_labels_test.go

```diff
@@ -100,7 +100,9 @@ func TestReconcileInterruptibleNodeLabel(t *testing.T) {
 	g.Expect(env.Create(ctx, cluster)).To(Succeed())
 	g.Expect(env.Create(ctx, node)).To(Succeed())
 	g.Expect(env.Create(ctx, infraMachine)).To(Succeed())
-	g.Expect(env.Create(ctx, machine)).To(Succeed())
+	// Note: We have to DeepCopy the machine, because the Create call clears the status and
+	// reconcileInterruptibleNodeLabel requires .status.nodeRef to be set.
+	g.Expect(env.Create(ctx, machine.DeepCopy())).To(Succeed())
```

Comment:
Note: this was an intended change in controller-runtime: https://github.com/kubernetes-sigs/controller-runtime/pull/1640